### DISCUSSION TITLE
Generate key log file

### DISCUSSION THREAD

**Discussion by octaviandumitran** [tlsfuzzer/tlslite-ng#522] [Q&A]
Can you add functionality to generate a key log file?
This should be something similar to what Python3.8+ does when SSLKEYLOGFILE is defined.

Alternatively, could you explain how I can generate the file myself?
Presumably, I would need to call keyingMaterialExporter, but I do not know exactly how to serialize that.

**Comment by tomato42** [comment]
I generally don't use it (I simply insert print statements when I need to inspect stuff), but I can clearly see how it would be useful, so I'm fine with having support for it.

generating it is a bit more involved that just using the `keyingMaterialExporter` as that one is for a different purpose. You need to export the master secret and the client random at a minimum (the available fields are documented in a proposed RFC: https://datatracker.ietf.org/doc/html/draft-ietf-tls-keylogfile-02 ). As to how to create it? likely some code in the `tlsconnection` or `tlsrecordlayer` modules that checks for that environment variable and then dumps the relevant key material into the file pointed to by the environment variable.

I've created #523 issue to track it.
